lib/rspec/matchers/built_in/contain_exactly.rb

```diff
@@ -6,6 +6,19 @@ module BuiltIn
       # Provides the implementation for `contain_exactly` and `match_array`.
       # Not intended to be instantiated directly.
       class ContainExactly < BaseMatcher
+        def initialize(expected=nil)
+          super
+          @transitive = false
+        end
+
+        # @api public
+        # Specifies that elements contained in actual and expected
+        # obey transitivity.  This lets match run much faster.
+        def transitive
```

Comment:
Do you think it would be possible to infer this from what `expected` is? E.g. if there's a matcher there or a Regex, they won't be transitive. And imply transitivity otherwise?